### PR TITLE
Replace broken FAST API with new FAST suggest API

### LINKED ISSUE #3581: FAST vocabulary type ahead lookup in H2 should work
Currently this box in H2 behaves as a free text entry box. No type ahead search results appear when you type in the box.

`App 176307 output: W, [2024-07-22T09:40:55.455746 #176307]  WARN -- : [f24fa9fe-2590-42b5-ae17-5047e08d6a50] Autocomplete results for river returned 404`

### PR DESCRIPTION
**Pull request by mjgiarlo**
- [x] @amyehodge tests in a deployed environment
- [x] @arcadiafalcone reviews comments on this PR about changes in the new FAST API

# Why was this change made?

Fixes #3581

# How was this change tested?

CI

### CODE REVIEW

File: config/settings.yml [hunk 1]
```diff
@@ -65,7 +65,7 @@ authorization_workgroup_names:
 
 # See https://www.oclc.org/developer/develop/web-services/fast-api/assign-fast.en.html for more information
 autocomplete_lookup:
-  url: http://fast.oclc.org/fastIndex/select
+  url: http://fast.oclc.org/searchfast/fastsuggest
```
Comment (mjgiarlo) [inline config/settings.yml:68]: @amyehodge @arcadiafalcone Here's the new FAST API suggested by #3581 (https://www.oclc.org/developer/api/oclc-apis/fast-api/assign-fast.en.html). It's different from the former API, though. See my other comments for more.

File: app/controllers/fast_controller.rb [hunk 2]
```diff
@@ -20,55 +20,40 @@ def show
   private
 
   def lookups(query)
-    # Try first without wildcard.
```
Comment (mjgiarlo) [inline app/controllers/fast_controller.rb:23]: As far as I can tell from https://www.oclc.org/developer/api/oclc-apis/fast-api/assign-fast.en.html, there is no support for wildcard queries anymore.

File: app/controllers/fast_controller.rb [hunk 3]
```diff
@@ -20,55 +20,40 @@ def show
...
 
-  # rubocop:disable Metrics/AbcSize
-  def lookup(query, wildcard: false)
+  def lookup(query) # rubocop:disable Metrics/AbcSize
     resp = lookup_connection.get do |req|
-      req.params['q'] = wildcard ? "keywords:(#{query}*)" : "keywords:(#{query})"
+      req.params['query'] = query
+      req.params['queryIndex'] = 'suggestall'
+      req.params['queryReturn'] = 'idroot,suggestall,tag'
+      req.params['suggest'] = 'autoSubject'
```
Comment (mjgiarlo) [inline app/controllers/fast_controller.rb:34]: @arcadiafalcone @amyehodge @edsu These are my best guesses based on https://www.oclc.org/developer/api/oclc-apis/fast-api/assign-fast.en.html and #3581

File: app/controllers/fast_controller.rb [hunk 4]
```diff
@@ -20,55 +20,40 @@ def show
...
       # Requesting extra records to try to increase the likelihood that get left anchored matches
       # since results are sorted by usage.
       req.params['rows'] = num_records * 2
-      req.params['start'] = 0
-      req.params['version'] = '2.2'
-      req.params['indent'] = 'on'
-      req.params['fl'] = 'id,fullphrase,type'
```
Comment (mjgiarlo) [inline app/controllers/fast_controller.rb:47]: These don't appear to be supported anymore.

File: app/controllers/fast_controller.rb [hunk 5]
```diff
@@ -78,49 +63,25 @@ def num_records
...
+  TAG_TYPES = {
+    100 => 'person',
+    110 => 'organization',
+    111 => 'event',
+    151 => 'place',
+    130 => 'title',
+    155 => 'genre',
+    148 => 'time',
+    # NOTE: Missing: 'meeting' => 'conference'
   }.freeze
```
Comment (mjgiarlo) [inline app/controllers/fast_controller.rb:82]: @amyehodge @arcadiafalcone @edsu The results from the new API do not include a type, but do include a tag. OCLC provide some sample JavaScript code that suggest a MARC tag mapping like the above. Note that I could not find any evidence that events and meetings are separate in this dataset.

File: app/controllers/fast_controller.rb [hunk 6]
```diff
@@ -78,49 +63,25 @@ def num_records
...
-    'chronological' => 'time'
+  TAG_TYPES = {
+    100 => 'person',
+    110 => 'organization',
+    111 => 'event',
+    151 => 'place',
+    130 => 'title',
+    155 => 'genre',
+    148 => 'time'
+    # NOTE: Missing: 'meeting' => 'conference'
```
Comment (mjgiarlo) [inline app/controllers/fast_controller.rb:81]: @arcadiafalcone AFAICT, the new FAST API does not distinguish between event-type authorities and meeting-type authorities. Is this problematic? Did I get it wrong?

Comment (arcadiafalcone) [reply app/controllers/fast_controller.rb:81]: Going from the corresponding MARC tags, 111 should be meeting/conference and 147 should be event. The data set downloads and the web search both still make the distinction between the two types, so perhaps the API documentation is in error?

Some sample values:
Paris Peace Conference (1919-1920) http://id.worldcat.org/fast/1696461 is a meeting
World War (1914-1918) http://id.worldcat.org/fast/1180746 is an event

Comment (mjgiarlo) [reply app/controllers/fast_controller.rb:81]: Ah, awesome, thank you. 😄

File: app/controllers/fast_controller.rb (hunk 3 above)
Comment (mjgiarlo) [reply app/controllers/fast_controller.rb:34]: Confirmed by looking at our initial implementation in 2021.

**Review by justinlittman — APPROVED** [review]

**Review by aaron-collier — APPROVED** [review]

**Pull request merged**